src/ROUTES.ts

```diff
@@ -420,6 +430,16 @@ const ROUTES = {
         getRoute: (iouType: ValueOf<typeof CONST.IOU.TYPE>, transactionID: string, reportID: string) => `create/${iouType}/start/${transactionID}/${reportID}/scan` as const,
     },
 
+    MONEY_REQUEST_STATE_SELECTOR: {
+        route: 'moneyrequest/state',
+
+        getRoute: (state?: string, backTo?: string, label?: string) =>
+            `${getUrlWithBackToParam(`moneyrequest/state${state ? `?state=${encodeURIComponent(state)}` : ''}`, backTo)}${
+                // the label param can be an empty string so we cannot use a nullish ?? operator
+                // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+                label ? `${backTo || state ? '&' : '?'}label=${encodeURIComponent(label)}` : ''
+            }` as const,
+    },
```

Comment:
No. i think its not because of the url but because it needs to be in the same `StackNavigator `, otherwise when doing `Navaigation.goBack()`, it is not able to modify the url and just goes back to the previous page without adding `state` param

https://github.com/Expensify/App/blob/9ad7ca0ad76fda24c04b42cc352482a2554b7f8b/src/libs/Navigation/AppNavigator/ModalStackNavigators.tsx#L80

https://github.com/Expensify/App/blob/9ad7ca0ad76fda24c04b42cc352482a2554b7f8b/src/libs/Navigation/AppNavigator/ModalStackNavigators.tsx#L204